src/Cabal.hs

```diff
@@ -53,9 +53,10 @@ run cabal args = do
     : "--build-depends=QuickCheck"
     : "--build-depends=template-haskell"
     : ("--repl-options=-ghci-script=" <> script)
-    : "--with-compiler" : doctest
-    : "--with-hc-pkg" : ghcPkg
-    : args) >>= waitForProcess >>= exitWith
+    : args ++ [
+      "--with-compiler", doctest
+    , "--with-hc-pkg", ghcPkg
+    ]) >>= waitForProcess >>= exitWith
```

Comment:
Maybe add a comment if you intend to let `cabal` handle duplicated flags based on their order.